File: ApiService/Models/Supplier.cs
using ApiService.Models.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ApiService.Models {
    public class Supplier : ILoadentEntity {
        public virtual int Id { get; set; }
        [Required]
        public virtual string Name { get; set; }
        public virtual int AddressId { get; set; }
        [Required]
        public virtual Address Address { get; set; }
    }
}

File: AngularLogin3/ClientApp/src/app/_models/Supplier.ts

export interface SupplierInterface {
  name: string;
  address: string;
};

export class Supplier implements SupplierInterface {
  name: string;
  address: string;

  constructor(name: string, address: string) {
    this.name = name;
    this.address = address;
  }

}

